docs/source/content/supervised/knn.rst

```diff
@@ -2,32 +2,160 @@
 k-Nearest Neighbors
 ====================
 
-K-Nearest Neighbors (KNN) is a basic classifier for machine learning. So we are trying to identify what class an object is in. To do this we look at the closest points (neighbors) to the object and the class with the majority of neighbors will be the class that we identify the object to be in. The k is the number of nearest neighbors to the object. So if k = 1 then the class the object would be in is the class of the closest neighbor. Let's look at an example.
+.. contents::
+  :local:
+  :depth: 3
+
+Introduction
+-------------
+
+K-Nearest Neighbors (KNN) is a basic classifier for machine learning. 
+So, we are trying to identify what class an object is in. To do this we 
```

Comment:
Identify that you already have a labeled data set, and you're trying to classify other points into one of the categories.  Without the background knowledge, one might be confused as to how you would know any categories at all.